core/src/test/scala/org/apache/spark/scheduler/SparkListenerSuite.scala

```diff
@@ -142,6 +142,7 @@ class SparkListenerSuite extends SparkFunSuite with LocalSparkContext with Match
     stageInfo.submissionTime should be ('defined)
     stageInfo.completionTime should be ('defined)
     taskInfoMetrics.length should be {4}
+    sc.stop()
```

Comment:
OK then it's definitely on purpose. I agree with stopping the context of course. Can you make a JIRA please? I think you're aware of this. I am not sure how this relates to memory usage though.